tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,187 @@
+struct MetadataView {
+    name: String,
+    version: Version,
+}
+
+pub(crate) async fn run(args: &Args, publish_kit_args: &PublishKitArgs) -> Result<()> {
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let infra_config = InfraConfig::from_path_or_lock(&args.infra_config_path, false)
+        .context(error::ConfigSnafu)?;
+    trace!("Parsed infra config: {:?}", infra_config);
+    ensure!(infra_config.kit_vendors.is_some(), error::NoVendorsSnafu);
+
+    // Fetch the vendor container registry uri
+    let vendor_registry_uri = infra_config
+        .kit_vendors
+        .as_ref()
+        .unwrap()
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    info!(
+        "Found vendor container registry at uri: {}",
+        vendor_registry_uri
+    );
+
+    // Auto resolve the expected paths for the metadata file and kit contents archive
+    let kit_path = publish_kit_args.kit_path.as_path();
+    let metadata_file = kit_path.join("kit-metadata.json");
+    let metadata_file = metadata_file.as_path();
+    ensure!(
+        metadata_file.exists(),
+        error::NoMetadataSnafu {
+            path: metadata_file
+        }
+    );
+    let mut platforms = Vec::new();
+    for arch in ["amd64", "arm64"] {
+        let path = kit_path.join(format!("kit-{}.tar", arch));
+        if path.exists() {
+            platforms.push(format!("linux/{}", arch));
+        }
+    }
+    ensure!(
+        !platforms.is_empty(),
+        error::NoArchiveSnafu { path: kit_path }
+    );
+
+    // Fetch a view of the metadata
+    let metadata_blob = read(metadata_file).await.context(error::IoSnafu)?;
+    let metadata: MetadataView =
+        serde_json::from_slice(metadata_blob.as_slice()).context(error::DeserializationSnafu)?;
+    trace!("Parse kit metadata to view: {:?}", metadata);
+
+    info!(
+        "Found local kit with name {} and version {}",
+        metadata.name, metadata.version
+    );
+
+    // We only really need the version and name from the metadata here to build our publish uri
+    let target_uri = format!(
+        "{}/{}:{}",
+        vendor_registry_uri, metadata.name, metadata.version
+    );
+
+    // Use docker to build and push the kit base image
+    info!(
+        "Creating and pushing kit container via docker to {}",
+        &target_uri
+    );
+    let metadata_encoded =
+        base64::engine::general_purpose::STANDARD.encode(metadata_blob.as_slice());
+
+    let docker_file = format!(
+        r#"FROM scratch
+ARG TARGETARCH
+
+LABEL {}={}
+
+ADD kit-$TARGETARCH.tar /
+"#,
+        KIT_METADATA_KEY, metadata_encoded
+    );
+    let docker_file_path = kit_path.join("Dockerfile");
+    write(&docker_file_path, docker_file)
+        .await
+        .context(error::IoSnafu)?;
+    let tag = format!("--tag={}", target_uri);
+    let platform = format!("--platform={}", platforms.join(","));
```

Comment:
I don't think pubsys should assume that both architectures exist; it should handle the case where one or both are present, and error out if none are.

Reply:
It does not assume that at all if you look up above platforms is generated only by what artifacts exist

Reply:
I understand this more now